### DISCUSSION TITLE
egsinp command list

### DISCUSSION THREAD

**Discussion by Jaap1957** [nrc-cnrc/EGSnrc#815] [Questions] [answered]
Hi, I am new to EGSnrc, it is a great program !

Managed to get through the 'getting-started.pdf'.
Now i have an adapted 'slab.egsinp' file.

I try to simulate a camera in an electron microscope.
I want to measure the dose in a few voxels around a pencil beam of electrons hitting a 'direct electron' camera (read: some metal and Si layers). 
With the "  :start input loop:  " ... construct
I can easily address a set of voxels.
i learned the :start input loop: command from the egsinp examples.

But i want to do more. Learning by going through the examples is a slow process. 

Is there a list of commands used in the egsiinp files,
which defines the "language"  used in egsinp ?

Have a great 2022 !
Jaap

**Comment by rtownson** [comment, marked as answer]
The input terms for egs++ are all described here:
https://nrc-cnrc.github.io/EGSnrc/doc/pirs898/
specifically, look under `common input syntax`, `geometries` and `sources`. This covers the vast majority of the possible inputs.

Each application will have some specific inputs. Those are described on the application documentation page, e.g. for egs_chamber:
https://nrc-cnrc.github.io/EGSnrc/doc/pirs898/egs_chamber.html

The input loop command is one that is currently overlooked in the common syntax documentation, but it is described on the egs_chamber page (it used to only be available for egs_chamber).

**Comment by Jaap1957** [comment]
Thanks.

So i found something special with the iinput loop, i understand.

Comparing the time needed to get a real result with EGSnrc , MCNP4C or GEANT4  i see you guys have done a great job.

I needed only a few days to get through install, understand enough of the input files, and start tweaking the tutor7pp example.

**Reply by rtownson** [reply]
That's great to hear!
